ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/utils/NetNullableElementUtils.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * ******************************************************************************
+ *   Copyright 2014-2015 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+package com.spectralogic.ds3autogen.net.utils;
+
+import com.spectralogic.ds3autogen.net.generators.elementparsers.*;
+
+import static com.spectralogic.ds3autogen.net.NetHelper.toNetType;
+import static com.spectralogic.ds3autogen.utils.ConverterUtil.hasContent;
+import static com.spectralogic.ds3autogen.utils.Helper.capFirst;
+import static com.spectralogic.ds3autogen.utils.Helper.stripPath;
+
+/**
+ * Static utility functions that create Nullable Elements. These are used in
+ * the .net package type model generators to create the parse element .net code
+ */
+public final class NetNullableElementUtils {
+
+    /**
+     * Creates a Nullable Element from the provided parameters
+     */
+    public static NullableElement createNullableElement(
+            final String name,
+            final String type,
+            final String componentType,
+            final boolean nullable,
+            final String xmlTag,
+            final String encapsulatingXmlTag,
+            final boolean isAttribute) {
+        if (hasContent(componentType)) {
```

Comment:
But then I'll need to create another if statement for:
if (hasContent(componentType) && isEmpty(encapsulatingXmlTag))